osm2gtfs/creators/ci_abidjan/trips_creator_ci_abidjan.py

```diff
@@ -0,0 +1,209 @@
+# coding=utf-8
+
+import collections
+from datetime import timedelta, datetime
+import transporthours
+from transitfeed import ServicePeriod
+
+from osm2gtfs.creators.trips_creator import TripsCreator
+from osm2gtfs.core.helper import Helper
+from osm2gtfs.core.elements import Line
+
+
+class TripsCreatorCiAbidjan(TripsCreator):
+
+    _DAYS_OF_WEEK = [ 'monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday', 'sunday' ]
+    _DEFAULT_START_TIME = "05:00:00"
+    _DEFAULT_END_TIME = "22:00:00"
+    _DAY_ABBREVIATIONS = {
+            'monday': 'Mo',
+            'tuesday': 'Tu',
+            'wednesday': 'We',
+            'thursday': 'Th',
+            'friday': 'Fr',
+            'saturday': 'Sa',
+            'sunday': 'Su'
+        }
+    _DEFAULT_HEADWAY = 30 # in minutes
+
+    def _date_range(self, start, end):
+        return self._DAY_ABBREVIATIONS[start] + '-' + self._DAY_ABBREVIATIONS[end]
+
+    def _days_abbreviation_from_transport_hour(self, a_transport_hour):
+        service_days = [ day_name for day_name in self._DAYS_OF_WEEK if day_name in a_transport_hour and a_transport_hour[day_name]]
+
+        if not service_days:
+            print('Transport_hour missing service days. Assuming 7 days a week.')
+            service_days = self._DAYS_OF_WEEK
+
+        if collections.Counter(service_days) == collections.Counter(self._DAYS_OF_WEEK):
+            return self._date_range('monday', 'sunday')
+        if collections.Counter(service_days) == collections.Counter(self._DAYS_OF_WEEK[:5]):
+            return self._date_range('monday', 'friday')
+        if collections.Counter(service_days) == collections.Counter(self._DAYS_OF_WEEK[:6]):
+            return self._date_range('monday', 'saturday')
+        if collections.Counter(service_days) == collections.Counter(self._DAYS_OF_WEEK[-2:]):
+            return self._date_range('saturday', 'sunday')
+
+        return ','.join([self._DAY_ABBREVIATIONS[day_name] for day_name in service_days])
+
+    def _set_default_times(self, transport_hour):
+        transport_hour['start_time'] = self._DEFAULT_START_TIME
+        transport_hour['end_time'] = self._DEFAULT_END_TIME
+
+    def _init_service_period(self, feed, hour):
+        service_id = self._days_abbreviation_from_transport_hour(hour)
+        service_period = ServicePeriod(id=service_id)
+        service_period.SetStartDate(self.config['feed_info']['start_date'])
+        service_period.SetEndDate(self.config['feed_info']['end_date'])
+        for i,day_of_week in enumerate(self._DAYS_OF_WEEK):
+            if (day_of_week in hour and hour[day_of_week]):
+                service_period.SetDayOfWeekHasService(i)
+        feed.AddServicePeriodObject(service_period)
+        print("Created new service period " + service_id)
+        return service_period
+
+    def _process_transport_hours(self, feed, transport_hours_list):
```

Comment:
can we be more specific in the naming ? maybe `_group_hours_by_service_period`

Reply:
ok